src/NServiceBus.Core/Transports/Msmq/MsmqUtilities.cs

```diff
@@ -71,6 +71,34 @@ static MsmqAddress GetIndependentAddressForQueue(MessageQueue q)
             return headers;
         }
 
+        public static bool TryOpenQueue(string address, out MessageQueue messageQueue)
+        {
+            messageQueue = null;
+            var msmqAddress = MsmqAddress.Parse(address);
+            var path = msmqAddress.PathWithoutPrefix;
+            try
+            {
+                if (MessageQueue.Exists(path))
+                {
+                    messageQueue = new MessageQueue(path);
+                    return true;
+                }
+            }
+            catch (MessageQueueException mex)
+            {
+                if (msmqAddress.IsRemote && (mex.MessageQueueErrorCode == MessageQueueErrorCode.IllegalQueuePathName))
+                {
+                    return false;
+                }
+                if (mex.MessageQueueErrorCode == MessageQueueErrorCode.QueueExists)
+                {
+                    return false;
+                }
```

Comment:
If these conditions all return false, and directly after the catch block you also return false, what's the point of doing anything special inside the catch block?